flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/StateAssignmentOperation.java

```diff
@@ -464,6 +464,14 @@ private void collectPartionableStates(
 	private static void checkParallelismPreconditions(OperatorState operatorState, ExecutionJobVertex executionJobVertex) {
 		//----------------------------------------max parallelism preconditions-------------------------------------
 
+		if (operatorState.getMaxParallelism() < executionJobVertex.getParallelism()) {
+			throw new IllegalStateException("The state for task " + executionJobVertex.getJobVertexId() +
+				" can not be restored. The maximum parallelism " + operatorState.getMaxParallelism() +
```

Comment:
can you added braces around the (max)parallelism? The error message currently looks like this:

```
Caused by: java.lang.IllegalStateException: The state for task adf090656b210b1609ad3203d4ee7329 can not be restored. The maximum parallelism 128 of the restored state is lower than the configured parallelism 140. Please reduce the parallelism of the task to be lower or equal to the maximum parallelism.
```

But i think
```
Caused by: java.lang.IllegalStateException: The state for task adf090656b210b1609ad3203d4ee7329 can not be restored. The maximum parallelism (128) of the restored state is lower than the configured parallelism (140). Please reduce the parallelism of the task to be lower or equal to the maximum parallelism.
```
looks nicer.